### PR TITLE
Add infrastructure for checking project filesystem

### PR DESCRIPTION
**Pull request by per1234**
I needed this for one of my projects and thought it could be useful for Arduino tooling projects as well, so I submit a "template" here for the team's consideration.

I wasn't able to find packaged tools that provide these checks so I used shell commands for the purpose.

---

There are differences in the filename restrictions between operating systems. The use of filenames that are not valid one one operating system in the project will cause problems for contributors or users (e.g., not possible to check out the repository).

Tasks are added to check for non-portable filenames:

- Presence of characters prohibited by an operating system in filenames
- Use of filenames reserved by an operating system

Tasks are also added to check for problems with symbolic links ("symlinks") contained in the project:

- Broken symlinks
- Circular symlinks

A GitHub Actions workflow is included to run the tasks on any change to the project files in order to avoid the introduction of problems with the project filesystem, and periodically in order to catch breakage caused by external changes.

### CODE REVIEW

File: workflow-templates/check-files-task.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+A GitHub Actions workflow is included to run the tasks on any change to the project files in order to avoid the
+introduction of problems with the project filesystem, and periodically in order to catch breakage caused by external
+changes.
+```
+
+## PR message
+
+```markdown
+There are differences in the filename restrictions between operating systems. The use of filenames that are not valid one one operating system in the project will cause problems for contributors or users (e.g., not possible to check out the repository).
```
Comment (umbynos) [inline workflow-templates/check-files-task.md:72]:
```suggestion
There are differences in the filename restrictions between operating systems. The use of filenames that are not valid on one operating system in the project will cause problems for contributors or users (e.g., not possible to check out the repository).
```
Typo

**Review by umbynos — APPROVED** [review]
Hi Per, Overall LGTM.
I only have one question.
Why is it required to run this type of check periodically?

**Comment by per1234** [comment]
>Why is it required to run this type of check periodically?

I have learned that it is important to run every workflow periodically to catch breakage caused by external changes. If this is not caught, then the breakage only becomes apparent when a contributor submits a PR, which is a bad experience for them (because they assume the workflow run failure is caused by their changes), and makes extra work for the maintainer (because they have to explain to the contributor why the workflow run failed).

Examples of relevant external changes for this workflow:

- Changes to GitHub Actions runner handling of the workflow
- Runner operating system updates
- GitHub Actions action updates (the action versions are only pinned to the major version ref)
- [Task](https://taskfile.dev/) updates (Task has an integrated shell interpreter, so in addition to the taskfile handling, even the handling of shell commands in a task can be affected by an update)

**Comment by umbynos** [comment]
Yeah, you are right, thanks for the explanation. In this case LGTM

**Pull request merged**